docs/gameroom_walkthrough/act_1/behind_the_scenes/behind_scene_1.md

```diff
@@ -0,0 +1,132 @@
+## I-1. Behind scene 1: Alice logs into Acme's Gameroom UI
+
+When a user logs in, the user interface (UI) component of the Gameroom client
+application works with the Gameroom REST API to check the user's email address
+and password. Each Gameroom daemon stores the user credentials in a local
+PostgreSQL database; user passwords are hashed so that they remain secret.
+
+### I-1.1. Acme UI sends authorization request to Gameroom REST API
+
+![](./images/auth_login_acme1.svg)
+
+When Alice clicks Log in, the Acme Gameroom UI hashes the password, then sends
+an authorization request to the Acme Gameroom daemon, gameroomd. The request is
+handled by the Gameroom REST API, which is a part of gameroomd.
+
+``` json
+    POST /users/authenticate
+    {
+      "email": "alice@acme.com",
+      "hashedPassword": "8e066d41...d99ada0d"
+    }
+```
+
+The UI does not reveal the user's password to the REST API because the password
+is used to encrypt signing keys.
+
+### I-1.2. Gameroom REST API authorizes login
+
+![](./images/auth_login_acme2.svg)
+
+When the Acme Gameroom REST API receives the authorization request for Alice, it
+re-hashes the password sent from the browser and compares the email and hashed
+password to Alice’s entry in the Acme Gameroom daemon's local database. If they
+match, authentication was successful.
+
+The `gameroom_user` table in the Gameroom database has the following schema:
+
+``` sql
+    CREATE TABLE IF NOT EXISTS gameroom_user (
+      email TEXT PRIMARY KEY,
+      public_key TEXT NOT NULL,
+      encrypted_private_key TEXT NOT NULL,
+      hashed_password TEXT NOT NULL
+    );
+```
+
+Alice's public and private key pair was created during registration and was
+added to the Acme Gameroom database. The database has the following entry:
+
+| email | hashed_password | public_key | encrypted_private_key |
+| :---: | :---: | :---: | :---: |
+|alice@acme.com|56ec82cb...480cad32|0384781f...5a7e4998|{\"iv\":...cgXrm\"}|
```

Comment:
Not a blocker for this PR, but we need to find a way to disable automatic links for this and other example email addresses and URLs in the document.  Users should not be able to click these items (plus, the blue is distracting).

Reply:
According to this helpful person, the fix is to add <span> tags around the item that should not be a link. I have not tested this yet.

https://eureka.ykyuen.info/2014/07/04/markdown-disable-url-auto-url-link-in-github-flavored-markdown/

Reply:
Would definitely agree, wasn't sure of a way to take care of this in just markdown.

Reply:
Also, the text in the table is supposed to be fixed-width font (surrounded by back-ticks).